module2/src/com/epam/decomposition/NumbersAnalyzer.java

```diff
@@ -0,0 +1,247 @@
+package com.epam.decomposition;
+
+
+public class NumbersAnalyzer {
+    //task 1
+    public static int findLeastCommonMultiple(int firstNumber, int secondNumber) {
+        return ((firstNumber * secondNumber) / findGreatestCommonDivisor(firstNumber, secondNumber));
+    }
+
+    private static int findGreatestCommonDivisor(int firstNumber, int secondNumber) {
+        while (firstNumber != secondNumber) {
+            if (firstNumber > secondNumber) {
+                firstNumber = firstNumber - secondNumber;
+            } else {
+                secondNumber = secondNumber - firstNumber;
+            }
+        }
+        return firstNumber;
+    }
+
+    private static int findGreatestCommonDivisor(int firstNumber, int secondNumber, int thirdNumber, int fourthNumber) {
+        return findGreatestCommonDivisor((findGreatestCommonDivisor(firstNumber, secondNumber)),
+                (findGreatestCommonDivisor(thirdNumber, fourthNumber)));
+        }
+
+    //task 2
+    public static int findLeastCommonMultiple(int firstNumber, int secondNumber,
+                                                            int thirdNumber, int fourthNumber) {
+        return findGreatestCommonDivisor(firstNumber,secondNumber,thirdNumber,fourthNumber);
+    }
+
+    //task 5
+    public static int findSecondLargestNumber(int[] array,int indexNumber) {
+        boolean isSorted = false;
+        while (!isSorted) {
+            isSorted = true;
+            for (int i = 1; i < array.length; i++) {
+                if (array[i] < array[i - 1]) {
+                    int temp = array[i - 1];
+                    array[i - 1] = array[i];
+                    array[i] = temp;
+                    isSorted = false;
+                }
+            }
+        }
+        return array[array.length  - indexNumber];
```

Comment:
You did not understand me. You don't need to sort the entire array to find the second largest element in an array. You just need to run on it once.
When you sort an array, you have to do too many unnecessary steps.